src/harness/reference_models/move_list_calc/move_list.py

```diff
@@ -0,0 +1,582 @@
+    """
+
+    # Check consistency of DPA type against frequency range.
+    # If correct, get reference incumbent antenna height above sea/ground level
+    if dpa_type == 'co-channel offshore':
+        if low_freq>=LOW_FREQ_COCH_OFFSHORE and high_freq<=HIGH_FREQ_COCH_OFFSHORE:
+            h_inc_ant = REF_INC_ANTENNA_HT_COCH_OFFSHORE
+        else:
+            raise ValueError('Inconsistency between DPA type %s and frequency range [%s,%s]'
+                             % (dpa_type, low_freq, high_freq))
+    elif dpa_type == 'co-channel inland':
+        if low_freq>=LOW_FREQ_COCH_INLAND and high_freq<=HIGH_FREQ_COCH_INLAND:
+            h_inc_ant = REF_INC_ANTENNA_HT_COCH_INLAND
+        else:
+            raise ValueError('Inconsistency between DPA type %s and frequency range [%s,%s]'
+                             % (dpa_type, low_freq, high_freq))
+    elif dpa_type == 'out-of-band inland':
+        if high_freq<=HIGH_FREQ_OOB_INLAND:
+            h_inc_ant = REF_INC_ANTENNA_HT_OOB_INLAND
+        else:
+            raise ValueError('Inconsistency between DPA type %s and frequency range [%s,%s]'
+                             % (dpa_type, low_freq, high_freq))
+    else:
+        raise ValueError('Unknown DPA type %s. Valid type: co-channel offshore, '
+                         'co-channel inland, out-of-band inland.' % dpa_type)
+
+    # Index the grant requests in the same order in the registration_request and
+    # grant_request lists [1, 2,..., N].
+    grant_index = range(1, len(registration_request) + 1)
+
+    # Loop over each protection point and find the move list
+    M_temp = []
+
+    for point in protection_points:
+
+        Mc = []         # Initialize move list for current protection constraint c
+
+        # Assign values to the protection constraint
+        constraint = ProtectionConstraint(latitude=point.latitude,
+                                          longitude=point.longitude,
+                                          lowFrequency=low_freq,
+                                          highFrequency=high_freq)
+
+        # Identify Nc CBSD grants in the neighborhood of protection constraint c
+        Nc_grants = findGrantsInsideNeighborhood(registration_request, grant_request,
+                                                 grant_index, constraint, exclusion_zone, dpa_type)
+
+        if len(Nc_grants):  # Found CBSDs in the neighborhood
+
+            # Form the matrix of interference contributions
+            I, sorted_grant_index, bearing = formInterferenceMatrix(Nc_grants, constraint,
+                                                                    h_inc_ant, num_iter, dpa_type)
+
+            # Find the index (nc) of the grant in the ordered list of grants such that
+            # the protection percentile of the interference from the first nc grants is below
+            # the threshold for all azimuths of the receiver antenna.
+            nc = find_nc(I, bearing, threshold)
```

Comment:
I am not sure the current find_nc is appropriate for out of band inland DPA.
According to the spec R2-IPM-4: 
> Unless otherwise noted by NTIA, inland out-of-band DPAs (below 3500 MHz) are defined as the 3 km exclusion zone around the sites listed in TR 15-517 (check if an existing Reference in document). For inland out-of-band DPAs, the protection level applied at DPA points shall be -125 dBm/MHz, and the reference antenna characteristic is an isotropic antenna 5 m above ground-level.  

So we shall not simulate the rotating antenna with 1.5degree, but simply add all signals of all neighboring CBSD with a single 0dBi antenna gain.